In [7]:
FIX_INVALID_DATA_LINK = "data/3_fix_invalid_data.csv"
import pandas as pd

df = pd.read_csv(FIX_INVALID_DATA_LINK)
print(df.columns)

Index(['age', 'job', 'marital', 'education', 'default', 'housing', 'loan',
       'contact', 'month', 'day_of_week', 'duration', 'campaign', 'pdays',
       'previous', 'poutcome', 'emp.var.rate', 'cons.price.idx',
       'cons.conf.idx', 'euribor3m', 'nr.employed', 'y'],
      dtype='str')


In [8]:
TARGET_VARIABLE_NAME = "y"
print("---------Count------------")
print(df[TARGET_VARIABLE_NAME].value_counts())
print("---------Proportion------------")
print(df["y"].value_counts(normalize=True) * 100)

---------Count------------
y
no     36531
yes     4637
Name: count, dtype: int64
---------Proportion------------
y
no     88.736397
yes    11.263603
Name: proportion, dtype: float64


The target variable is highly imbalanced, with approximately 11.3% of customers subscribing to the new plan.